### DISCUSSION TITLE
Transfer tiers based on fifo and drive capacity (using tiers as 'cache')

### DISCUSSION THREAD

**Discussion by kaifastromai** [minio/minio#16210] [Q&A] [answered]
Forgive me if this has been mentioned elsewhere (I searched the issues and didn't seem to find anything), but I would like to create a 'hot' layer in minio (using high performance nvme's) that will automatically transfer the oldest items as storage becomes limited. 

I've read the documentation on life-cycle management, but it appears you can only schedule specific lengths to transfer for objects, instead of what I suggested above. Would it be better to implement this at the hardware level of each node using something like RAID, or is there something else that may enable this facility?

**Comment by klauspost** [comment, marked as answer]
There is no way to do what you want, and it is very unlikely there will be.

The main problem is that what you propose requires "complete knowledge" of a pool. MinIO can scale to billions of objects, so getting "complete knowledge" can be very time consuming and you bottleneck your entire tiering to a single process across the cluster.

Tiering must be able to determine if an object (version) should be tiered only by looking at the object itself. That is the only way to make these types of operations scalable.

**Reply by kaifastromai** [reply]
I know amazon s3 has something somewhat similar? Would it be possible to implement at the nodes themselves . From my understanding MinIO is particularly fond of RAID and the like, but could something similar be done by some other configuration?
